From b0c2d5a2660946f6c3b73346b84309f8a849b7c5 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Tue, 30 Oct 2018 18:18:57 +0200
Subject: [PATCH 234/741] sdk_dpaa: set the skb queue mapping when looping

commit 662a7c9e93c30793e9ca8bf142e1cbe2d661a9f9 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Save the current CPU ID on ingress, when FSL_DPAA_DBG_LOOP is set.
Use the skb_set_queue_mapping() call instead of skb_record_rx_queue()
because the stack isn't involved and won't compensate for the additional
offset.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
index b212755a0655..435e2a061452 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
@@ -575,6 +575,8 @@ static inline int dpa_skb_loop(const struct dpa_priv_s *priv,
 		return 0; /* loop disabled by default */
 
 	skb_push(skb, ETH_HLEN); /* compensate for eth_type_trans */
+	/* Save the current CPU ID in order to maintain core affinity */
+	skb_set_queue_mapping(skb, raw_smp_processor_id());
 	dpa_tx(skb, dpa_loop_netdevs[priv->loop_to]);
 
 	return 1; /* Frame Tx on the selected interface */
-- 
2.17.1

